### PR TITLE
add support for loading OME-TIFFs

### PR DESCRIPTION
**Pull request by tlnagy**
See more discussion in https://github.com/JuliaIO/FileIO.jl/issues/144

### CODE REVIEW

**Comment by tlnagy** [comment]
This needs to wait till OMETIFF.jl is registered in Metadata.jl and new tag is pushed.

**Comment by tlnagy** [comment]
ffc.pcx is released under a CC0 license from here: http://www.fileformatcommons.com/pcx-file-format/

File: src/registry.jl [hunk 1]
```diff
@@ -154,6 +147,21 @@ add_format(format"FLAC","fLaC",".flac",[:FLAC])
 
 ### Complex cases
 
+# Handle OME-TIFFs, which are identical to normal TIFFs with the primary difference being the filename and embedded XML metadata
+tiff_magic = (UInt8[0x4d,0x4d,0x00,0x2a], UInt8[0x4d,0x4d,0x00,0x2b], UInt8[0x49,0x49,0x2a,0x00],UInt8[0x49,0x49,0x2b,0x00])
```
Comment (timholy) [inline src/registry.jl]: Add `const` in front of this? Doing so should make `detecttiff` faster.

File: test/query.jl [hunk 2]
```diff
@@ -286,16 +286,20 @@ end
...
-    q = query(joinpath(file_dir, "magic1.tiff"))
-    @test typeof(q) == File{format"TIFF"}
-    q = query(joinpath(file_dir, "magic2.tiff"))
-    @test typeof(q) == File{format"TIFF"}
+    q = query(joinpath(file_dir, "ffc.pcx"))
+    @test typeof(q) == File{format"PCX"}
     open(q) do io
         @test position(io) == 0
         skipmagic(io)
-        @test position(io) == 4
```
Comment (timholy) [inline test/query.jl]: Now that each function has its own type, we could change how `skipmagic` works:
```julia
skipmagic(io, ::typeof(detect_ometiff)) = seek(io, 4)
skipmagic(io, ::typeof(detect_noometiff)) = seek(io, 4)
```
and then I think the original test would pass?

File: src/registry.jl (hunk 1 above)
Comment (tlnagy) [inline src/registry.jl]: Sure thing!

File: test/query.jl (hunk 2 above)
Comment (tlnagy) [inline test/query.jl]: Where would it be best to put these `skipmagic` definitions? In `registry.jl` or `query.jl` where skipmagic is defined.

Comment (timholy) [inline test/query.jl]: These specializations probably have to go in `registry.jl`, since we need `detect_ometiff` to be defined already.

**Comment by tlnagy** [comment]
Hey @timholy, I made the changes you suggested. I think the failures on nightly aren't related to this PR.

**Comment by timholy** [comment]
This is great, thanks!

**Pull request merged**